### PR TITLE
DOC: Backport doc updates about supported configurations

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Backports this PR to the 2025.2 release branch: https://github.com/uxlfoundation/scikit-learn-intelex/pull/2286

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

Not applicable.

**Performance**

Not applicable.

### CODE REVIEW

File: doc/sources/oneapi-gpu.rst [hunk 1]
```diff
@@ -12,65 +12,76 @@
...
-|intelex| supports oneAPI concepts, which
-means that algorithms can be executed on different devices: CPUs and GPUs.
-This is done via integration with
-`dpctl <https://intelpython.github.io/dpctl/latest/index.html>`_ package that
-implements core oneAPI concepts like queues and devices.
+|intelex| can execute computations on different devices (CPUs, GPUs) through the SYCL framework in oneAPI.
+
+The device used for computations can be easily controlled through the target offloading functionality (e.g. through ``sklearnex.config_context(target_offload="gpu")`` - see rest of this page for more details), but for finer-grained controlled (e.g. operating on arrays that are already in a given device's memory), it can also interact with objects from package |dpctl|, which offers a Python interface over SYCL concepts such as devices, queues, and USM (unified shared memory) arrays.
+
+While not strictly required, package |dpctl| is recommended for a better experience on GPUs.
```
Comment (icfaust) [inline doc/sources/oneapi-gpu.rst:26]: I believe this statement to be too general (and I know this is a backport). I think listing why dpctl would be recommended is a good idea (starting point may be https://intelpython.github.io/dpctl/latest/beginners_guides/managing_devices.html#creating-sub-devices). From my perspective (and this is an opinion) what dpctl provides is deeper control of SYCL-derived devices and operation, not necessarily a 'better experience'.

Comment (icfaust) [reply doc/sources/oneapi-gpu.rst:26]: Also not sure why the link to dpctl is removed, unless that exists elsewhere.

Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst:26]: You could create another pull request and then we can backport both commits.

Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst:26]: > Also not sure why the link to dpctl is removed, unless that exists elsewhere.

What do you mean removed?

File: doc/sources/oneapi-gpu.rst [hunk 2]
```diff
@@ -12,65 +12,76 @@
...
 
-  1. The :code:`target_offload` option can be used to set the device primarily
-     used to perform computations. Accepted data types are :code:`str` and
-     :code:`dpctl.SyclQueue`. If you pass a string to :code:`target_offload`,
-     it should either be ``"auto"``, which means that the execution
-     context is deduced from the location of input data, or a string
-     with SYCL* filter selector. The default value is ``"auto"``.
-
-  2. The :code:`allow_fallback_to_host` option
+  1. The :code:`target_offload` argument (in ``config_context`` and in ``set_config`` / ``get_config``)
```
Comment (icfaust) [inline doc/sources/oneapi-gpu.rst:58]: While I know this is in the GPU section, there should be a warning here that device offload is only possible with a dpc build as this is not directly obvious, and make cause user confusion. This aspect of the code will fail in those circumstances, even when target_offload="cpu"

Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst:58]: Yes, could be good to add to the docs. But starting with 2025.0, there aren't any non-dpc builds in public channels anymore.

File: doc/sources/oneapi-gpu.rst [hunk 3]
```diff
@@ -83,16 +94,27 @@ call :code:`sklearnex.get_config()`.
...
+  device.
+
+  .. note::
+    All the input data for an algorithm must reside on the same device.
+
+  .. warning::
+    The :code:`usm_ndarray` can only be consumed by the base methods
+    like :code:`fit`, :code:`predict`, and :code:`transform`.
+    Note that only the algorithms in |intelex| support
+    :code:`usm_ndarray`. The algorithms from the stock version of |sklearn|
```
Comment (icfaust) [inline doc/sources/oneapi-gpu.rst:110]: Not strictly correct. Dpctl tensor support exists for sklearn as an experimental feature when array_api is enabled, which uses the array_namespace and only exists for some algorithms.  Next release of dpnp (the other variant of usm_ndarray) will support in next release.

**Comment by david-cortes-intel** [comment]
@icfaust Could you add the suggestions in a new PR to the main branch?

File: doc/sources/quick-start.rst [hunk 4]
```diff
@@ -12,30 +12,29 @@
...
 ####################
 
-Get ready to elevate your scikit-learn code with |intelex| and experience the benefits of accelerated performance in just a few simple steps.
+Get ready to elevate your |sklearn| code with |intelex| and experience the benefits of accelerated performance in just a few simple steps.
 
 Compatibility with Scikit-learn*
 ---------------------------------
 
-Intel(R) Extension for Scikit-learn is compatible with the last four versions of scikit-learn.
+|intelex| is compatible with the latest stable releases of |sklearn| - see :ref:`software-requirements` for more details.
```
Comment (icfaust) [inline doc/sources/quick-start.rst:26]: I do worry about the inconsistency in the docs in using 'intelex' vs 'Intel(R) Extension for Scikit-learn' vs 'scikit-learn-intelex' etc. Please make a ticket as a follow up to make consistent our use of these names.

**Comment by icfaust** [comment]
> @icfaust Could you add the suggestions in a new PR to the main branch?

@david-cortes-intel please make a ticket and leave it unassigned

File: doc/sources/oneapi-gpu.rst [hunk 5]
```diff
@@ -102,11 +124,11 @@ An example on how to patch your code with Intel CPU/GPU optimizations:
...
-               [8., 7.], [8., 8.], [25., 80.]], dtype=np.float32)
+                 [8., 7.], [8., 8.], [25., 80.]], dtype=np.float32)
    with config_context(target_offload="gpu:0"):
       clustering = DBSCAN(eps=3, min_samples=2).fit(X)
 
 
-.. note:: Current offloading behavior restricts fitting and inference of any models to be
-     in the same context or absence of context. For example, a model trained in the GPU context with
-     target_offload="gpu:0" throws an error if the inference is made outside the same GPU context.
+.. note:: Current offloading behavior restricts fitting and predictions (a.k.a. inference) of any models to be
```
[the remark this answers is not in the document]
Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst:132]: @icfaust This is intentional. "inference" typically means something different in statistical contexts, where sklearn may also be used.

[the remark this answers is not in the document]
Comment (icfaust) [reply doc/sources/oneapi-gpu.rst:132]: https://en.wikipedia.org/wiki/List_of_Latin_phrases_(I)#id_est From a grammatical standpoint, use of 'a.k.a' is informal. i.e. would be a better fit given the context

File: doc/sources/oneapi-gpu.rst (hunk 2 above)
Comment (icfaust) [reply doc/sources/oneapi-gpu.rst:58]: That would imply that no one may be using a custom ARM build on a regular basis (which I suspect will change).

File: doc/sources/oneapi-gpu.rst [hunk 6]
```diff
@@ -12,65 +12,76 @@
...
 .. _oneapi_gpu:
 
 ##############################################################
 oneAPI and GPU support in |intelex|
 ##############################################################
 
-|intelex| supports oneAPI concepts, which
-means that algorithms can be executed on different devices: CPUs and GPUs.
-This is done via integration with
-`dpctl <https://intelpython.github.io/dpctl/latest/index.html>`_ package that
```
Comment (icfaust) [inline doc/sources/oneapi-gpu.rst:24]: @david-cortes-intel the link that I mentioned, sorry, the comment probably should be here

File: doc/sources/oneapi-gpu.rst (hunk 1 above)
Comment (icfaust) [reply doc/sources/oneapi-gpu.rst:26]: https://github.com/uxlfoundation/scikit-learn-intelex/pull/2315#discussion_r1952518911

File: doc/sources/oneapi-gpu.rst (hunk 6 above)
Comment (david-cortes-intel) [reply doc/sources/oneapi-gpu.rst:24]: @icfaust The link is handled through the new entry in the intersphinx config.

You can verify that it renders in the artifacts from the CI job that builds docs:
image

**Review by ethanglaser — APPROVED** [review]
Its just a backport so any review should go towards https://github.com/uxlfoundation/scikit-learn-intelex/pull/2286, if there are additional suggestions as a follow-up to that then create a ticket

**Comment by ethanglaser** [comment]
@Alexsandruss @napetrov have merge capabilities

**Comment by david-cortes-intel** [comment]
Release is already out there so not relevant anymore.
